identity/src/commonMain/kotlin/com/android/identity/asn1/ASN1BitString.kt

```diff
@@ -0,0 +1,92 @@
+package com.android.identity.asn1
+
+import kotlinx.io.bytestring.ByteStringBuilder
+
+class ASN1BitString(
+    val numUnusedBits: Int,
+    val value: ByteArray
+): ASN1PrimitiveValue(tag = TAG_NUMBER) {
+
+    constructor(booleanValues: BooleanArray):
+            this(
+                if (booleanValues.size == 0) 0 else (8 - (booleanValues.size % 8)),
+                encodeBooleans(booleanValues)
+            )
+
+    override fun encode(builder: ByteStringBuilder) {
+        ASN1.appendUniversalTagEncodingLength(builder, tag, enc, value.size + 1)
+        builder.append(numUnusedBits.toByte())
+        builder.append(value)
```

Comment:
The standard requires unused bits to be set to 0, so there's only one possible encoding. I don't see anything here that guarantees that, if the caller used the ByteArray constructor directly. Either add a check()/require() or force those values to 0 before appending?